src/core/agent/agent.ts

```diff
@@ -172,68 +174,122 @@ class Agent {
     );
   }
 
-  async start(): Promise<void> {
+  async start(agentUrls: Partial<AgentUrls>): Promise<void> {
     if (!Agent.isOnline) {
-      await this.storageSession.open(walletId);
-      this.basicStorageService = new BasicStorage(
-        this.getStorageService<BasicRecord>(this.storageSession)
-      );
-      this.identifierStorage = new IdentifierStorage(
-        this.getStorageService<IdentifierMetadataRecord>(this.storageSession)
-      );
-      this.credentialStorage = new CredentialStorage(
-        this.getStorageService<CredentialMetadataRecord>(this.storageSession)
-      );
-      this.connectionStorage = new ConnectionStorage(
-        this.getStorageService<ConnectionRecord>(this.storageSession)
-      );
-      this.connectionNoteStorage = new ConnectionNoteStorage(
-        this.getStorageService<ConnectionNoteRecord>(this.storageSession)
-      );
-      this.notificationStorage = new NotificationStorage(
-        this.getStorageService<NotificationRecord>(this.storageSession)
-      );
-
+      const exitsUrls = await this.getAgentUrl();
+      if (
+        (!exitsUrls.url || !exitsUrls.bootUrl) &&
+        (!agentUrls.url || !agentUrls.bootUrl)
+      ) {
+        //  @TODO
+        throw new Error("Require KERI URL and Boot URL to start the agent.");
+      }
       await signifyReady();
       const bran = await this.getBran();
-      // @TODO - foconnor: Review of Tier level.
-      this.signifyClient = new SignifyClient(
-        ConfigurationService.env.keri.keria.url,
-        bran,
-        Tier.low,
-        ConfigurationService.env.keri.keria.bootUrl
-      );
 
-      this.agentServicesProps = {
-        signifyClient: this.signifyClient,
-        eventService: new EventService(),
-      };
+      if (agentUrls.url && !exitsUrls.url) {
+        // @TODO - foconnor: Review of Tier level.
+        this.signifyClient = new SignifyClient(
+          agentUrls.url,
+          bran,
+          Tier.low,
+          agentUrls.bootUrl
+        );
+        await this.signifyClient.boot();
+        await this.signifyClient.connect();
```

Comment:
`start` should only call `connect`.

We need a separate function to call `boot` once that can be triggered by the UI during onboarding.